base/MAPL_ObsUtil.F90

```diff
@@ -565,9 +580,34 @@ function get_filename_from_template_use_index (obsfile_start_time, obsfile_inter
 
     ! parse time info
     !
-    call fill_grads_template ( filename, file_template, &
-         experiment_id='', nymd=nymd, nhms=nhms, _RC )
-    inquire(file= trim(filename), EXIST = exist)
+    !
+    allow_wild_char=.true.
+    !
+    j= index(file_template, '*')
+    if (j>0) then
+       ! wild char exist
+       !!print*, 'pos of * in template =', j
+       file_template_left = file_template(1:j-1)
+       call fill_grads_template ( filename_left, file_template_left, &
+            experiment_id='', nymd=nymd, nhms=nhms, _RC )
```

Comment:
Won't `fill_grads_template()` conveniently ignore the `*` at this step?   I.e., can't we fill the template before the `if (j > 0)`?

Reply:
I ask because the logic can be greatly simplified if so.

Reply:
Oh, I see you are right. Yes, we can fill the template while ignoring the `*`. 
I had  a concern (or a misconception) that `fill_grads_template` can not handle `*`, which leads me to chop the string into two pieces then bind them together.  
I have changed the code to make the logic clean.  Thank you!